ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -20,32 +35,106 @@ class BaseLine extends React.Component {
   render() {
     const props = this.props
     const { label, cursor, geometryKey, unitKey, widthKey } = props
-
     return (
-      <div className="input-location">
-        <TextField
-          label={label}
-          value={this.state.value}
-          onChange={value => {
-            this.setState({ value })
-            const fn = cursor(geometryKey)
-            try {
-              fn(JSON.parse(value))
-            } catch (e) {}
-          }}
-        />
-        <Units value={props[unitKey]} onChange={cursor(unitKey)}>
+      <React.Fragment>
+        <div className="input-location">
           <TextField
-            type="number"
-            label="Buffer width"
-            min={0.000001}
-            value={props[widthKey]}
-            onChange={cursor(widthKey)}
+            label={label}
+            value={this.state.value}
+            onChange={value => {
+              this.setState({ value })
+              const fn = cursor(geometryKey)
+              try {
+                fn(JSON.parse(value))
+              } catch (e) {}
+            }}
+            onBlur={() => this.isValidInput(this.state.value)}
+            onFocus={value => {
+              this.isValid = true
+              this.setState()
+            }}
           />
-        </Units>
-      </div>
+          <Units value={props[unitKey]} onChange={cursor(unitKey)}>
+            <TextField
+              type="number"
+              label="Buffer width"
+              min={0.000001}
+              value={props[widthKey]}
+              onChange={cursor(widthKey)}
+            />
+          </Units>
+        </div>
+        {this.isValid ? (
+          ''
+        ) : (
+          <Invalid>
+            &nbsp;
+            <span className="fa fa-exclamation-triangle" />
+            &nbsp; {this.inValidMessage}
+          </Invalid>
+        )}
+      </React.Fragment>
     )
   }
+  isValidInput(value) {
+    this.inValidMessage = ''
+    this.inValidPoint = ''
+    if (this.isValidPolygon(value)) {
+      this.isValid = true
+      this.setState({ value })
+    } else {
+      this.isValid = false
+      this.setState({ value })
+    }
+  }
+  is2DArray(coordinates) {
+    try {
+      let parsedCoords = JSON.parse(coordinates)
+      return Array.isArray(parsedCoords) && Array.isArray(parsedCoords)
+    } catch (e) {
+      return false
+    }
+  }
+  isValidListOfPoints(coordinates) {
```

Comment:
❓ it would be better for this function to be purely functional with no side effects. One possible method would be to rename this validateListOfPoints, and throw an error when the list is not valid. Then maybe in the render method, you would handle the logic to set the invalid message by using a try catch block. I think it would be good to rename/refactor isValidInput and isValidPolygon as well.

Reply:
Agree with @samuelechu ☝️ the fewer side effects (preferably zero) the better.